feat(stats-detectors): Add initial endpoint regression issue docs (#8427)
getsentry/sentry-docs: src/docs/product/issues/issue-details/performance-issues/endpoint-regression/index.mdx

## Detection Criteria

The detector for performance issues periodically checks the 95th percentile transaction duration of the most common endpoints in your project. When a significant increase in the p95 value is detected and has been sustained for some time, a regression issue is created.

## Regression Evidence

To find additional info, go to the Issues page and click on the Endpoint Regression issue you're interested in. In the top section of the "Details" tab, you'll find the following:

- **Endpoint Name:** The name of the transaction that has regressed.
- **Change in Duration:** The value of absolute and relative change in duration.
- **Approximate Start Time:** The approximate time when the regression occurred.

## Charts

The below chart shows the p95 transaction duration as well as the transaction throughput over a period of up to 14 days before and 14 days after the regression was identified.

> ![Endpoint Regression Issue Chart](endpoint-regression-chart.png)

Reviewer: The throughput chart is moving to the side bar. Can we remove mentions of it from here and update the screenshot once the changes are in?

## Potential Causes

The "Potential Causes" section shows a list of spans that may have contributed to the slowdown the most. Next to each span, you'll find the following:

- Span description that leads to the [Span Summary](/product/performance/transaction-summary/) page where you can find more information about that specific span.
- Throughput of span before and after the observed change in duration.

> - P95 of span self time before and after the observed change in duration.

Reviewer: This table has changed so we likely want to update it to reflect the new content. You can checkout the function regression issue docs for a reference.

Author: done 👍

---
title: "Endpoint Regression"
sidebar_order: 41
description: "Learn more about Endpoint Regression issues and how to diagnose and fix them."
---

<Include name="feature-available-for-user-group-early-adopter.mdx" />

Endpoint Regression issues capture a generic class of problems where the duration of a transaction has increased over time. This can lead to degraded performance of your application. Sentry proactively monitors some of your most common endppoints out of the box and reports any possible regressions under the Endpoint Regression issue type.

## Detection Criteria

The detector for performance issues periodically checks the 95th percentile transaction duration of the most common endpoints in your project. When a significant increase in the p95 value is detected and has been sustained for some time, a regression issue is created.

## Regression Evidence

> To find additional information about your regressed endpoint, go to the top section which shows the following:

Author: added more details

- **Endpoint Name:** The name of the transaction that has regressed.
- **Change in Duration:** The value of absolute and relative change in duration.
- **Approximate Start Time:** The approximate time when the regression occurred.

## Charts

> 
> ![Endpoint Regression Issue Chart](endpoint-regression-chart.png)
> 
> The chart shows a window of up to 14 days before and after the regression was identified. Here, you can see the p95 transaction duration as well as the transaction throughput.

Reviewer 2: ```suggestion

The below chart shows the p95 transaction duration as well as the transaction throughput over a period of up to 14 days before and 14 days after the regression was identified.

![Endpoint Regression Issue Chart](endpoint-regression-chart.png)

```

## Potential Causes

> The "Potential Causes" section shows a list of spans that might have contributed to the slow down the most. Next to each span, you can find the following:
> 

Reviewer 2: ```suggestion
The "Potential Causes" section shows a list of spans that may have contributed to the slowdown the most. Next to each span, you'll find the following:

```

> - Span description that leads to the [Span Summary](/product/performance/transaction-summary/) page where you can find more data about that specific span.
> - Throughput of that span before and after the observed change in duration.
> - P95 of that span self time before and after the observed change in duration.

Reviewer 2: ```suggestion
- Span description that leads to the [Span Summary](/product/performance/transaction-summary/) page where you can find more information about that specific span.
- Throughput of span before and after the observed change in duration.
- P95 of span self time before and after the observed change in duration.
```

## Compare Events

> The "Compare Events" section shows example events before and after the regression took place. Baseline events are events that have duration close to the original baseline P95 duration and regressed events have duration close to the new p95 baseline.
> The "Compare Events" section lets you compare example events from before and after the regression occurred. We consider events to be part of the "Baseline events" when their duration is close to the original P95 duration. We refer to the events that have a duration close to the newest P95 baseline as "Regressed events".

Reviewer 2: ```suggestion
The "Compare Events" section lets you compare example events from before and after the regression occurred. We call events that are close to the original P95 baseline duration "Baseline events", and events that are close to the new P95 baseline "Regressed events".
```
